### PR TITLE
Restructuring so we can package devenv

### PR DESCRIPTION
**Pull request by IanWoodard**
Restructured the repo to enable packaging and use of virtual environments. These changes make it easier to install and use the devenv utilities and enable the use of virtual environments to manage dependencies and isolate the environment. We still need to update the install-devenv.sh script to use the venv.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+name = "sentry_devenv"
+version = "0.0.1"
+authors = [
+  { name="Joshua Li", email="joshua.li@sentry.io" },
+  { name="Ian Woodard", email="ian.woodard@sentry.io" },
+  { name="Buck Evan", email="buck.evan@sentry.io" },
+]
+description = "Utilities for setting up a Sentry development environment"
+readme = "README.md"
+requires-python = ">=3.9"
```
Comment (joshuarli) [inline pyproject.toml:15]: should be 3.11

**Review by joshuarli — CHANGES_REQUESTED** [review]
You need to fix `install-devenv.sh` to make a venv and pip install this, and put that in PATH where it's needed.

```
devenv installed at /Users/runner/.local/share/sentry-devenv/bin/devenv
./install-devenv.sh: line 68: devenv: command not found
```

**Comment by IanWoodard** [comment]
> You need to fix `install-devenv.sh` to make a venv and pip install this, and put that in PATH where it's needed.
> 
> ```
> devenv installed at /Users/runner/.local/share/sentry-devenv/bin/devenv
> ./install-devenv.sh: line 68: devenv: command not found
> ```

#26 should take care of this

**Comment by joshuarli** [comment]
superseded by https://github.com/getsentry/devenv/pull/26

**Pull request closed without merging**